packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -110,75 +173,80 @@ const Indicator = () => {
-    const isOnPrettyUrl = prettyUrlRegex.test(host)
-    const { siteInfo, currentBuild, latestBuild } = await getBuildInfo()
+      // currentBuild is the most recent build that is not QUEUED.
+      // latestBuild is the most recent build that finished running (ONLY status ERROR or SUCCESS)
+      const isOnPrettyUrl = prettyUrlRegex.test(host)
+      const { siteInfo, currentBuild, latestBuild } = await getBuildInfo()
 
-    if (!buildId) {
-      if (isOnPrettyUrl || host === `localhost`) {
-        buildId = latestBuild?.id
-      } else {
-        // Match UUID from preview build URL https://build-af44185e-b8e5-11eb-8529-0242ac130003.gtsb.io
-        const buildIdMatch = host?.match(/build-(.*?(?=\.))/)
-        if (buildIdMatch) {
-          buildId = buildIdMatch[1]
+      if (!buildId) {
+        if (isOnPrettyUrl || host === `localhost`) {
+          buildId = latestBuild?.id
+        } else {
+          // Match UUID from preview build URL https://build-af44185e-b8e5-11eb-8529-0242ac130003.gtsb.io
+          const buildIdMatch = host?.match(/build-(.*?(?=\.))/)
+          if (buildIdMatch) {
+            buildId = buildIdMatch[1]
+          }
         }
       }
-    }
 
-    const newBuildInfo = {
-      currentBuild,
-      latestBuild,
-      siteInfo,
-      isOnPrettyUrl,
-    }
-
-    if (currentBuild?.buildStatus === BuildStatus.BUILDING) {
-      // Keep status as up to date builds where we cannot fetch manifest id info.
-      setBuildInfo({ ...newBuildInfo, buildStatus: BuildStatus.UPTODATE })
-    } else if (currentBuild?.buildStatus === BuildStatus.ERROR) {
-      setBuildInfo({ ...newBuildInfo, buildStatus: BuildStatus.ERROR })
-    } else if (buildId && buildId === newBuildInfo?.currentBuild?.id) {
-      setBuildInfo({ ...newBuildInfo, buildStatus: BuildStatus.UPTODATE })
-    } else if (
-      buildId &&
-      buildId !== newBuildInfo?.latestBuild?.id &&
-      currentBuild?.buildStatus === BuildStatus.SUCCESS
-    ) {
-      if (refreshNeeded) {
-        setBuildInfo({ ...newBuildInfo, buildStatus: BuildStatus.SUCCESS })
-      } else {
-        const { hasPageChanged, errorMessage } = await hasPageDataChanged(
-          buildId
-        )
+      const newBuildInfo = {
+        currentBuild,
+        latestBuild,
+        siteInfo,
+        isOnPrettyUrl,
+      }
 
-        if (errorMessage) {
-          setBuildInfo({
-            ...newBuildInfo,
-            buildStatus: BuildStatus.ERROR,
-            errorMessage,
-          })
-        } else if (hasPageChanged) {
-          // Force a "This page has updated message" until a page is refreshed
-          refreshNeeded = true
-          // Build updated, data for this specific page has changed!
+      if (currentBuild?.buildStatus === BuildStatus.BUILDING) {
+        setBuildInfo({ ...newBuildInfo, buildStatus: BuildStatus.BUILDING })
+      } else if (
+        currentBuild?.buildStatus === BuildStatus.ERROR &&
+        !usingContentSync
```

Comment:
correct me if im wrong, but within the info button, we check and do a different flow if usingContentSync is present- is this check necessary?

Reply:
On second thought you're right 👍 while we don't use the build status actions when content sync is running, we do use the build status to determine when we should start counting to time out so we should be setting the build error status